src/utils/utils.ts

```diff
@@ -27,22 +27,36 @@ export function getOrCreateTerminal(): vscode.Terminal {
     return globalState.maestroTerminal;
 }
 
-export async function updateYamlSchemaAssociations(schemaPath: string) {
+/**
+ * Updates the YAML schema associations in VS Code settings to ensure only our schema is used
+ * for Maestro YAML files. This prevents conflicts with other YAML schemas.
+ * Note: This function only manages the main extension schema patterns and does not affect
+ * test-specific schema configurations.
+ */
+export async function updateYamlSchemaAssociations(schemaPath: string = './schema/schema.v0.json') {
     const yamlConfig = vscode.workspace.getConfiguration('yaml');
-
     const currentSchemas = yamlConfig.get<{ [key: string]: string[] }>('schemas') || {};
-
     const filePatterns = getFilePatterns();
 
+    // Create updated schemas by removing ALL entries that point to our schema file
     const updatedSchemas = Object.fromEntries(
-        Object.entries(currentSchemas).filter(
-            ([, patterns]) => !patterns.some(pattern => filePatterns.includes(pattern))
+        Object.entries(currentSchemas).filter(([schema]) => {
+            // Remove any schema entry that points to our schema file, regardless of the path
+            return !schema.endsWith('schema.v0.json');
+        })
+    );
+
+    // First, remove any schema registrations from global settings
```

Comment:
Is this a user-hostile change? What if the user wants global config, because all of their projects are the same?

Reply:
Initially, we effect the change on the global configuration.

We read the file patterns then write to global configuration. Right now, we have cleared the global and applied the file patterns to workspace only.

Reply:
Thinking about this, we should consider a better approach.